### ISSUE TITLE
Clean up when launch fails

### ISSUE THREAD

**Issue by yuvipanda** [jupyterhub/binderhub#245]
If Binder fails to launch a JupyterHub server, it should try to explicitly stop it and not leave pods randomly lying around orphaned. https://github.com/jupyterhub/kubespawner/issues/95 is related

**Comment by choldgraf** [comment]
It's hard for me to tell if this is fixed now, though I'll note that we have a decent amount of failed launches, and to my knowledge this doesn't result in orphaned pods. Can we call this fixed?

**Comment by minrk** [comment]
I don't believe that we actively attempt top stop the server. We have gotten better at failing to proceed if the request stops, for instance, so some orphaned server cases (e.g. user disconnected before launch started) are less likely to occur now.
